vllm/worker/model_runner.py

```diff
@@ -869,11 +905,31 @@ def _maybe_cupy_nccl():
         yield
 
 
+def _pad_to_alignment(x: List[int], multiple_of: int, pad: int) -> List[int]:
+    return x + [pad] * ((-len(x)) % multiple_of)
+
+
 def _pad_to_max(x: List[int], max_len: int, pad: int) -> List[int]:
     assert len(x) <= max_len
     return x + [pad] * (max_len - len(x))
 
 
+def _make_tensor_with_pad_for_alignment(
+    x: List[int],
+    pad: int,
+    dtype: torch.dtype,
+    device: Optional[Union[str, torch.device]],
+) -> torch.Tensor:
+    """Create a tensor of a given list x with padding.
+    It adds paddings to align with graph batch size. See
+    _get_graph_batch_size for more details.
+    """
+    batch_size = len(x)
+    batch_size = _get_graph_batch_size(batch_size)
```

Comment:
I've been testing this branch in eager mode and I'm getting `CUBLAS_STATUS_EXECUTION_FAILED` errors which disappear if I comment out this line of code. I think eager mode isn't handled correctly currently (some tensors/lists are padded and others not).

Reply:
Let me check whether I can reproduce with new commits, and will provide more details if it persists.

Reply:
Seems to be resolved by latest commits (tested with both `SHOULD_ALIGN=False` and `SHOULD_ALIGN=True`).

Reply:
Nice! That's great to hear that